velox/exec/LocalPartition.h

```diff
@@ -109,6 +103,7 @@ class LocalExchangeSource {
   std::vector<VeloxPromise<bool>> producerPromises_;
   int pendingProducers_{0};
   bool noMoreProducers_{false};
+  bool isFinished_{false};
```

Comment:
General question : I understand what this variable is doing. But why was it not needed before ?

Reply:
There was a bug where the task with a local exchange would get stuck if a limit above the exchange would finish early. See https://github.com/facebookincubator/velox/pull/1332